pkg/store/consul/pcstore/consul_store_test.go

```diff
@@ -1015,6 +1020,126 @@ func TestWatchAndSync(t *testing.T) {
 	}
 }
 
+func TestWatchAndSyncWithDelete(t *testing.T) {
+	store := consulStoreWithFakeKV()
+	quit := make(chan struct{})
+	defer close(quit)
+	syncSignal := make(chan struct{})
+	defer close(syncSignal)
+	deleteSignal := make(chan struct{})
+	defer close(deleteSignal)
+
+	example := examplePodCluster()
+	pc1, err := store.Create(
+		example.PodID,
+		example.AvailabilityZone,
+		"name1",
+		example.PodSelector,
+		example.Annotations,
+		consultest.NewSession(),
+	)
+	if err != nil {
+		t.Fatalf("Couldn't create test pod cluster: %s", err)
+	}
+
+	// Include two pod clusters as initial, but we only created one. we
+	// should see a delete call for the second
+	syncer := &RecordingSyncer{
+		InitialClusters: []fields.ID{pc1.ID, "some_other_ID"},
+		SyncSignal:      syncSignal,
+		DeleteSignal:    deleteSignal,
+	}
+
+	go func() {
+		err := store.WatchAndSync(syncer, quit)
+		if err != nil {
+			t.Fatalf("Couldn't start WatchAndSync(): %s", err)
```

Comment:
funny story - `Fatalf` calls `FailNow`, and https://golang.org/pkg/testing/#B.FailNow "FailNow must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test."

Now, I'm not sure what happens if some other goroutine calls it. Perhaps we could try?